spec/smarty_streets_spec.rb

```diff
@@ -3,67 +3,88 @@
 describe SmartyStreets do
   let(:auth_id) { 'MYAUTHID' }
   let(:auth_token) { 'MYAUTHTOKEN' }
+  let(:request_read_timeout) { 5 }
+  let(:request_open_timeout) { 5 }
   let(:candidates) { 1 }
-  let(:response) { double(:response, code: response_code, body: response_body) }
-  let(:response_code) { }
-  let(:response_body) { }
-
 
   before do
     SmartyStreets.configure do |c|
       c.auth_id    = auth_id
       c.auth_token = auth_token
       c.candidates = candidates
+      c.request_read_timeout = request_read_timeout
+      c.request_open_timeout = request_open_timeout
     end
 
-    allow(HTTParty).to receive(:get).and_return(response)
+    http = double
+    allow(Net::HTTP).to receive(:start).and_yield http
+    allow(http).to receive(:request)
+      .with(an_instance_of(Net::HTTP::Get))
+      .and_return(Net::HTTPResponse)
   end
 
   context SmartyStreets::Request do
     context 'unauthorized request' do
-      let(:response_code) { 401 }
+      before do
+        allow(Net::HTTPResponse).to receive(:code)
+          .and_return(401)
+      end
 
       specify 'raises an InvalidCredentials error' do
-        expect {
-          locations = SmartyStreets.standardize {}
-        }.to raise_error SmartyStreets::Request::InvalidCredentials
+        expect do
+          SmartyStreets.standardize {}
+        end.to raise_error SmartyStreets::Request::InvalidCredentials
       end
     end
 
     context 'malformed data' do
-      let(:response_code) { 400 }
+      before do
+        allow(Net::HTTPResponse).to receive(:code)
+          .and_return(400)
+      end
 
-      specify 'raises an InvalidCredentials error' do
-        expect {
+      specify 'raises a MalformedData error' do
+        expect do
           SmartyStreets.standardize {}
-        }.to raise_error SmartyStreets::Request::MalformedData
+        end.to raise_error SmartyStreets::Request::MalformedData
       end
     end
 
     context 'payment required' do
-      let(:response_code) { 402 }
+      before do
+        allow(Net::HTTPResponse).to receive(:code)
+          .and_return(402)
+      end
 
-      specify 'raises an InvalidCredentials error' do
-        expect {
+      specify 'raises a PaymentRequired error' do
+        expect do
           SmartyStreets.standardize {}
-        }.to raise_error SmartyStreets::Request::PaymentRequired
+        end.to raise_error SmartyStreets::Request::PaymentRequired
       end
     end
 
     context 'remote server error' do
-      let(:response_code) { 500 }
+      before do
+        allow(Net::HTTPResponse).to receive(:code)
+          .and_return(500)
+      end
 
-      specify 'raises an InvalidCredentials error' do
-        expect {
+      specify 'raises an RemoteServerError error' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.